### PR TITLE
Testing framework fixes

### PR DESCRIPTION
**Pull request by mwichmann**
- Test framework now uses a subdirectory named "scons" below the base temporary directory. This gives something invariant to tell antivirus software to ignore without having to exclude the tmpdir itself.

- MSVS "live" tests of project files (e.g. `test/MSVS/vs-14.3-exec.py`) adjusted to look for the generated executable with an `.exe` suffix. This had recently started failing on the line:

  `test.run(program=test.workpath('sub dir', 'foo'), stdout="foo.c\n")`

  as somehow `sub dir\foo` was created, and then `foo.exe` was not matched causing the binary not to execute and failing the test.

## Contributor Checklist:

* [X] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [X] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

**Comment by mwichmann** [comment]
This is example new output of a test that was being rejected by antivirus software:
```
1/1 (100.00%) C:\Users\mats\AppData\Local\Programs\Python\Python313\python.exe test\CPPDEFINES\live.py
FAILED test of C:\Users\mats\Documents\github\scons\scripts\scons.py
        at line 774 of C:\Users\mats\Documents\github\scons\testing\framework\TestCommon.py (run)
        from line 458 of C:\Users\mats\Documents\github\scons\testing\framework\TestSCons.py (run)
        from line 79 of C:\Users\mats\Documents\github\scons\test\CPPDEFINES\live.py
        OSError(22, 'Operation did not complete successfully because the file contains a virus or potentially unwanted software', None, 225, None)
```

**Comment by mwichmann** [comment]
Hmmm, while the tests worked just fine on both Linux and Windows locally, I hadn't run the tests of the framework itself. That presumably needs adjusting for new path, I didn't think of that.

**Comment by mwichmann** [comment]
No.... it's not the path.  Instead, it's partly that I break an assumption of the framework tests: they check for the case of a `FileNotFoundError` in case the command to run isn't found.  The framework itself doesn't do anything with this outcome, until I made the change to fail the test, so we have a difference between *expected* command-missing errors (framework tests) and *unexpected* command-missing. The change takes the framework tests down a path they don't expect (ending up in the `sys.exit` caused by calling `fail_test`) and they get unhappy.

There are also some new failures there introduced by 3.13, which I'll have to investigate separately.

**Comment by mwichmann** [comment]
I wonder if it's better to migrate the fail-on-`OSError` up to `TestSCons`?

**Comment by mwichmann** [comment]
Going to update this, to remove the exception-handling part of the fix in `TestCmd`. It trips over what happens in `TestCommon.start()`. That's the place the framework tries to handle exceptions, I just don't think it does a good job of it, but you can't really paper around what a child class does in the parent class. It may still be that handling in `TestSCons`, the child of `TestCommon`, can work. My first guess did not).

File: test/MSVS/vs-14.1-exec.py [hunk 1]
```diff
@@ -20,20 +22,18 @@
...
 # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-#
-
-__revision__ = "__FILE__ __REVISION__ __DATE__ __DEVELOPER__"
 
 """
 Test that we can actually build a simple program using our generated
-Visual Studio 14.1 project (.vcxproj) and solution (.sln) files
+Visual Studio 14.0 project (.vcxproj) and solution (.sln) files
```
Comment (bdbaddog) [inline test/MSVS/vs-14.1-exec.py:28]: Shouldn't this be 14.1 and not 14.0 ? (same in the 14.2 test)

Comment (mwichmann) [reply test/MSVS/vs-14.1-exec.py:28]: I've lost the context (will try to mentally resync later) but I think there's been a common file format for several vs releases. hard to describe this... too many version descriptions involved.

**Pull request merged**